src/Confluent.Kafka/Serialization/StringDeserializer.cs

```diff
@@ -26,32 +30,93 @@ public class StringDeserializer : IDeserializer<string>
         Encoding encoding;
 
         /// <summary>
+        ///     Name of the encoding configuration parameter for the case of key deserialization.
+        /// </summary>
+        public const string KeyEncodingConfigParam = "dotnet.string.deserializer.encoding.key";
+
+        /// <summary>
+        ///     Name of the encoding configuration parameter for the case of value deserialization.
+        /// </summary>
+        public const string ValueEncodingConfigParam = "dotnet.string.deserializer.encoding.value";
+
+
+        /// <summary>
         ///     Initializes a new StringDeserializer class instance.
         /// </summary>
         /// <param name="encoding">
-        ///     The encoding to use when deserializing.
+        ///     The encoding to use when deserializing. For available options, refer to:
+        ///     https://msdn.microsoft.com/en-us/library/system.text.encoding(v=vs.110).aspx
         /// </param>
         public StringDeserializer(Encoding encoding)
         {
             this.encoding = encoding;
         }
 
         /// <summary>
+        ///     Initializes a new StringDesrializer class instance.
+        ///     The encoding encoding to use must be provided via <see cref="Consumer" /> configuration properties.
```

Comment:
oh, I put the note about available encodings on the wrong constructor - meant this to be a note on the other one.